ci_framework/roles/rhol_crc/tasks/configuration.yml

```diff
@@ -0,0 +1,30 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Get RHOL/CRC pull secret
+  ansible.builtin.copy:
+    dest: "{{ rhol_crc_pull_secret_path }}"
+    content: |
+      {{ rhol_crc_pull_secret_path }}
+
+- name: Set RHOL/CRC configuration options
```

Comment:
The way I found to iterate and set the configuration values based on the dictionary you mentioned is the following one:

```
- name: Set RHOL/CRC configuration options
  ansible.builtin.command: "{{ cifmw_rhol_crc_binary }} config set {{ item.key }} {{ item.value }}"
  with_dict: "{{ cifmw_rhol_crc_config_defaults | ansible.builtin.combine(cifmw_rhos_crc_config) }}"
  register: configuration_result
```
It means that the configuration dictionary keys should be the accepted directives of the `crc config set command`:

```
[2023-03-10 02:44:08] {afuscoar@afuscoar} (~)$ -> /home/afuscoar/.Redhat/Projects/github/ci-framework/bin/crc config set --help
Sets a crc configuration property.
CONFIG-KEYS: 

* consent-telemetry                          Consent to collection of anonymous usage data (yes/no)
* cpus                                       Number of CPU cores (must be greater than or equal to '2')
```

So the dictionary:

```
cifmw_rhol_crc_config_defaults:
  consent-telemetry: "no"
  cpus: 5
```
I've tested and seems to work. Let me know what do you think.

Reply:
looks OK. As said, for later, we may want to create our own module for that, it would avoid having to call a loop. Care to add some TODO in the file, for later reference?